### PR TITLE
toText replaced with toString in DateTime and TimeSpan

### PR DESCRIPTION
**Pull request by marcusnaslund**
Not used in other projects, so doesn't break the API. :)
Also fixes `replaceAll` in `String`, it used to leak badly.

### CODE REVIEW

File: source/base/DateTime.ooc [hunk 1]
```diff
@@ -79,24 +79,43 @@ DateTime: cover {
...
+		second free()
+		second = result replaceAll("%h", hourString)
+		result free()
+		result = second replaceAll("%m", minuteString)
+		second free()
+		second = result replaceAll("%s", secondString)
+		result free()
+		result = second replaceAll("%zzz", millisecond3String)
+		second free()
+		second = result replaceAll("%z", millisecondString)
```
Comment (marcusnaslund) [inline source/base/DateTime.ooc]: Yes this is ugly. We might consider implementing some `replaceAllAndFreeOriginal` methods (with a better name of course) and similarly for some other `String` methods... what do you think @sebastianbaginski @thomasfanell ?

Comment (thomasfanell) [inline source/base/DateTime.ooc]: Is there a strong reason why we can't have `replaceAll` destroy the original `String`?

Comment (marcusnaslund) [inline source/base/DateTime.ooc]: Consistency, I'd say, no other method destroys the original (I think?).

Comment (thomasfanell) [inline source/base/DateTime.ooc]: Would a default `Bool` be prettier (defaults to false)? If true, then free the original. That would cater to the existing style and just add the option of an automatic free.

Comment (marcusnaslund) [inline source/base/DateTime.ooc]: That works for me. I'd like that for `print` and `println` as well.

Comment (marcusnaslund) [inline source/base/DateTime.ooc]: Let's be happy with this for now, I will create an issue to discuss this further.

Comment (marcusnaslund) [inline source/base/DateTime.ooc]: #1763

**Comment by marcusnaslund** [comment]
Peer review @sebastianbaginski ?

File: source/base/DateTime.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/base/DateTime.ooc]: > what do you think @sebastianbaginski

I think we need to move away from ooc asap.

**Pull request merged**